core/services/registrysyncer/syncer.go

```diff
@@ -7,27 +7,23 @@ import (
 	"sync"
 	"time"
 
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	capabilitiespb "github.com/smartcontractkit/chainlink-common/pkg/capabilities/pb"
 	"github.com/smartcontractkit/chainlink-common/pkg/services"
 	"github.com/smartcontractkit/chainlink-common/pkg/types"
 	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
 
 	kcr "github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/capabilities_registry"
 	"github.com/smartcontractkit/chainlink/v2/core/logger"
 	p2ptypes "github.com/smartcontractkit/chainlink/v2/core/services/p2p/types"
 	evmrelaytypes "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
 )
 
-type HashedCapabilityID [32]byte
-type DonID uint32
-
-type State struct {
-	IDsToDONs         map[DonID]kcr.CapabilitiesRegistryDONInfo
-	IDsToNodes        map[p2ptypes.PeerID]kcr.CapabilitiesRegistryNodeInfo
-	IDsToCapabilities map[HashedCapabilityID]kcr.CapabilitiesRegistryCapabilityInfo
-}
-
 type Launcher interface {
-	Launch(ctx context.Context, state State) error
+	Launch(ctx context.Context, registry *LocalRegistry) error
```

Comment:
typically we want `accept interfaces return structs`. why is this accepting a struct while LocalRegistiry is implementing an interface?

Reply:
We can't use an interface here unfortunately because registry has data that we need to read. We could expose those members via getters but that seems clunky to me.